private/buf/buflsp/server.go

```diff
@@ -215,9 +215,22 @@ func (s *server) Formatting(
 		return nil, err
 	}
 
+	nodeInfo := file.fileNode.NodeInfo(file.fileNode)
 	return []protocol.TextEdit{
 		{
-			Range:   infoToRange(file.fileNode.NodeInfo(file.fileNode)),
+			Range: protocol.Range{
+				// The formatter always returns the entire file; make sure we
+				// represent the edit as the entire range (which may not match up to
+				// the file's NodeInfo.Start()).
```

Comment:
This comment doesn't actually explain what's going on here precisely. The way I would propose you do this is as follows:
```go
range_ := infoToRange(file.FileNode.NodeInfo(file.fileNode))
// NOTE: This is working around a limitation with the current compiler's parser. There are
// cases where leading comments are not included in the overall file node's span. To remedy
// this, we flatten 
range_.Start.Line = 0
range_.Start.Character = 0
```